components/director/internal/systemfetcher/loader.go

```diff
@@ -175,25 +205,135 @@ func (d *DataLoader) upsertAppTemplates(ctx context.Context, appTemplateInputs [
+			}
+			log.C(ctx).Infof("Successfully created application template with id: %q", templateID)
+			continue
+		}
+
+		var appInput1 map[string]interface{}
+		if err := json.Unmarshal([]byte(appTmplInput.ApplicationInputJSON), &appInput1); err != nil {
+			return errors.Wrapf(err, "while unmarshaling application input json")
+		}
+		var appInput2 map[string]interface{}
+		if err := json.Unmarshal([]byte(appTemplate.ApplicationInputJSON), &appInput2); err != nil {
+			return errors.Wrapf(err, "while unmarshaling application input json")
+		}
+
+		if !reflect.DeepEqual(appInput1, appInput2) {
+			log.C(ctx).Infof("Updating application template with id %q", appTemplate.ID)
+			appTemplateUpdateInput := model.ApplicationTemplateUpdateInput{
+				Name:                 appTmplInput.Name,
+				Description:          appTmplInput.Description,
+				ApplicationNamespace: appTmplInput.ApplicationNamespace,
+				ApplicationInputJSON: appTmplInput.ApplicationInputJSON,
+				Placeholders:         appTmplInput.Placeholders,
+				AccessLevel:          appTmplInput.AccessLevel,
+			}
+			if err := d.appTmplSvc.Update(ctx, appTemplate.ID, appTemplateUpdateInput); err != nil {
+				return errors.Wrapf(err, "while updating application template with id %q", appTemplate.ID)
 			}
-			log.C(ctx).Infof(fmt.Sprintf("Successfully registered application template with id: %q", templateID))
+			log.C(ctx).Infof("Successfuly updated application template with id %q", appTemplate.ID)
 		}
 	}
 
 	return nil
 }
 
+func enrichApplicationTemplateInput(appTemplateInputs []model.ApplicationTemplateInput) []model.ApplicationTemplateInput {
+	var enriched []model.ApplicationTemplateInput
+	for _, appTemplateInput := range appTemplateInputs {
+		if appTemplateInput.Description == nil {
+			appTemplateInput.Description = str.Ptr(appTemplateInput.Name)
+		}
+
+		if appTemplateInput.Placeholders == nil || len(appTemplateInput.Placeholders) == 0 {
+			appTemplateInput.Placeholders = []model.ApplicationTemplatePlaceholder{
+				{
+					Name:        "name",
+					Description: str.Ptr("Application’s technical name"),
+				},
+				{
+					Name:        "display-name",
+					Description: str.Ptr("Application’s display name"),
+				},
+			}
+		}
+
+		if appTemplateInput.AccessLevel == "" {
+			appTemplateInput.AccessLevel = model.GlobalApplicationTemplateAccessLevel
+		}
+
+		if appTemplateInput.Labels == nil {
+			appTemplateInput.Labels = map[string]interface{}{managedAppProvisioningLabelKey: false}
+		}
+		enriched = append(enriched, appTemplateInput)
+	}
+	return enriched
+}
+
+func enrichWithIntegrationSystemIDLabel(applicationInputJSON, intSystemID string) (string, error) {
+	var appInput map[string]interface{}
+
+	if err := json.Unmarshal([]byte(applicationInputJSON), &appInput); err != nil {
+		return "", errors.Wrapf(err, "while unmarshaling application input json")
+	}
+
+	labels, ok := appInput[labelsJSONKey]
```

Comment:
Integration system ID is a top level property of the app, not a label, maybe this is a mistake here?